include/fpu.h

```diff
@@ -45,105 +45,105 @@ void FPU_ESC6_EA(Bitu func,PhysPt ea);
 void FPU_ESC7_Normal(Bitu rm);
 void FPU_ESC7_EA(Bitu func,PhysPt ea);
 
-
-typedef union {
-    double d;
+union FPU_Reg {
+	double d = 0.0;
+	struct {
 #ifndef WORDS_BIGENDIAN
-    struct {
-        uint32_t lower;
-        int32_t upper;
-    } l;
+		uint32_t lower;
+		int32_t upper;
 #else
-    struct {
-        int32_t upper;
-        uint32_t lower;
-    } l;
+		int32_t upper;
+		uint32_t lower;
 #endif
-    int64_t ll;
-} FPU_Reg;
+	} l;
+	int64_t ll;
+};
 
-typedef struct {
-    uint32_t m1;
-    uint32_t m2;
-    uint16_t m3;
+struct FPU_P_Reg {
+	uint32_t m1 = 0;
+	uint32_t m2 = 0;
+	uint16_t m3 = 0;
 
-    uint16_t d1;
-    uint32_t d2;
-} FPU_P_Reg;
+	uint16_t d1 = 0;
+	uint32_t d2 = 0;
+};
 
-enum FPU_Tag {
+enum FPU_Tag : uint8_t {
 	TAG_Valid = 0,
 	TAG_Zero  = 1,
 	TAG_Weird = 2,
 	TAG_Empty = 3
 };
 
-enum FPU_Round {
-	ROUND_Nearest = 0,		
+enum FPU_Round : uint8_t {
+	ROUND_Nearest = 0,
 	ROUND_Down    = 1,
-	ROUND_Up      = 2,	
+	ROUND_Up      = 2,
 	ROUND_Chop    = 3
 };
 
-typedef struct FPU_rec {
-	FPU_Reg		regs[9];
-	FPU_P_Reg	p_regs[9];
-	FPU_Tag		tags[9];
-	uint16_t		cw,cw_mask_all;
-	uint16_t		sw;
-	uint32_t		top;
-	FPU_Round	round;
-} FPU_rec;
-
-#define L2E		1.4426950408889634
-#define L2T		3.3219280948873623
-#define LN2		0.69314718055994531
-#define LG2		0.3010299956639812
+struct FPU_rec {
+	FPU_Reg regs[9]      = {};
+	FPU_P_Reg p_regs[9]  = {};
+	FPU_Tag tags[9]      = {};
+	uint16_t cw          = 0;
+	uint16_t cw_mask_all = 0;
+	uint16_t sw          = 0;
+	uint32_t top         = 0;
+	FPU_Round round      = {};
+};
 
+constexpr auto L2E = 1.4426950408889634;
+constexpr auto L2T = 3.3219280948873623;
+constexpr auto LN2 = 0.69314718055994531;
+constexpr auto LG2 = 0.3010299956639812;
```

Comment:
While you're at it, can get sane names for these? Took me almost 10 mins to reverse engineer them 😅 

```
constexpr auto Log2_E = 1.4426950408889634;
constexpr auto Ln10DivLn2 = 3.3219280948873623;
constexpr auto Ln2 = 0.69314718055994531;
constexpr auto Log10_2 = 0.3010299956639812;
```

Reply:
Added! `cmath` had 'em too :-) 

Switched to the built-ins as `constexpr`s, for the two that involved dividing log 10 / 2 and ln 10 / 2.